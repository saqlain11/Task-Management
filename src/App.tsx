import React from "react";
import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import NotFound from "pages/NotFound";
import { TaskTable } from "components";

const App: React.FC = () => {
  return (
    <Router>
      <Routes>
        <Route index element={<TaskTable />}></Route>
        <Route path="*" element={<NotFound />} />
      </Routes>
    </Router>
  );
};

export default App;
