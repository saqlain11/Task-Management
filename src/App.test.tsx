import { render, waitFor } from "@testing-library/react";
import App from "./App";
import React, {lazy, Suspense} from "react";
import { act } from "react-dom/test-utils";
jest.mock("react",()=>({
  ...jest.requireActual("react"),
   lazy :() => ()=>"Component"
  }))

describe("App Suite", () => {
  it("should Render Correctly", async() => {

      const { asFragment } =  render(<Suspense fallback="loading"><App /></Suspense>);
      await waitFor(()=> expect(asFragment()).toMatchSnapshot())
  });
});
