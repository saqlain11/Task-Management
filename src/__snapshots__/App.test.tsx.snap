// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Suite should Render Correctly 1`] = `
<DocumentFragment>
  Component
</DocumentFragment>
`;
